draft-ietf-jsonpath-base.md

```diff
@@ -253,19 +254,15 @@ Unicode Scalar Value:
   E000 to 10FFFF. JSON values of type string are sequences of Unicode scalar values.
 
 Singular Path:
-: A JSONPath expression built from segments which each produce at most one node.
+: A JSONPath expression built from segments which each, regardless of the input value,
+  produces at most one node.
```

Comment:
I can't quite figure out whether this should be "produce" (segments which) or "produces" (each).

Reply:
The intention is that "produces" binds to "each". Otherwise "which each" would be left dangling.

We could reword to "... segments, each of which, regardless of the input value, produces at most one node." if that helps?

Reply:
I made that change as it's better English.